### ISSUE TITLE
Document how HTML parsing works 

### ISSUE THREAD

**Issue by RobertDober** [pragdave/earmark#357]
Spinoff from #356 

The following shall go into the documentation:

Also check why oneline tags are broken? Did I miss a regression in #353?

------
Here is what kind of HTML Earmark supports, and I will update the documentation which is not good (was even missing lately)

- Oneline HTML Tags

```markdown
   <tag...>{content}</tag>{suffix}
```
which will render
```elixir
   {"tag", [], ["content"], %{verbatim: true}} # 1.4.6 format
```

- One level of a block

```markdown
<tag>
    {content}
</tag>
```

```elixir
{"tag", [], [content], %{verbatim: true}}
```

**Comment by sreyansjain** [comment]
Earmark.as_html!("<mark>Test</mark>")
produces
"<p>\n  &lt;mark&gt;Test&lt;/mark&gt;\n</p>\n"

Earmark.as_html!("<span>Test</span>")
produces
"<p>\n  &lt;span&gt;Test&lt;/span&gt;\n</p>\n"

It was fine in 1.4.4

**Comment by sreyansjain** [comment]
In 1.4.4 Tweet embedding as is worked. But now its getting incorrectly parsed in 1.4.5

You can try with this
```
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Destiny does not happen without action.</p>&mdash; Mikael Cho (@mikaelcho) <a href="https://twitter.com/mikaelcho/status/1275454780819685377?ref_src=twsrc%5Etfw">June 23, 2020</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
```

**Comment by RobertDober** [comment]
@sreyansjain thank you for your comment

I do not think the current behavior or 1.4.4 is what we want, we want a oneline tag to be parsed as HTML

Concerning your Tweet example, that will definitely not work in 1.4.* but hopefully in 1.5 (c.f. #358)

**Comment by RobertDober** [comment]
Also document how comments are parsed #361

**Comment by lud** [comment]
Hello, 

I found that

```
    """
    <meta>
    key value</meta>


    Hello
    """
```

Parses as

```
[
  {"meta", [], ["key value</meta>", "", "", "Hello"],
   %{meta: %{verbatim: true}}}
]
```

**Comment by RobertDober** [comment]
This will be fixed in 1.4.10
